signing/vrfsigner.go

```diff
@@ -0,0 +1,34 @@
+package signing
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/spacemeshos/ed25519"
+)
+
+// VRFSigner is a signer for VRF purposes
+type VRFSigner struct {
+	privKey []byte
+}
+
+// Sign signs a message for VRF purposes
+func (s VRFSigner) Sign(m []byte) ([]byte, error) {
+	return ed25519.Sign(s.privKey, m), nil
+}
+
+// NewVRFSigner creates a new VRFSigner from a 32-byte seed
+func NewVRFSigner(seed []byte) (*VRFSigner, []byte, error) {
+	if len(seed) < ed25519.SeedSize {
+		return nil, nil, fmt.Errorf("seed must be >=%d bytes (len(seed)=%d)", ed25519.SeedSize, len(seed))
+	}
+	vrfPub, vrfPriv, err := ed25519.GenerateKey(bytes.NewReader(seed))
+	if err != nil {
+		return nil, nil, err
+	}
+	return &VRFSigner{privKey: vrfPriv}, vrfPub, nil
+}
+
+// VRFVerify verifies a message and signature, given a public key
+func VRFVerify(msg, sig, pub []byte) (bool, error) {
```

Comment:
Why does this method need to return an error in its signature? Does it need to conform to some interface?

Reply:
Sort of, the BLS verification method returned an error and the code around it expects it. I'd remove this, but this is a temporary implementation (we need to replace the standard ED with some unique-signature variant) and while I guess the final implementation will also not need to return an error, I'm not 100% sure, so I prefer to defer this change to when we know for sure.

But now that you mention it - I'll add a TODO.

Reply:
On second thought, I'll just change the signature now, it's easy and not worth the TODO.